src/ap/java/org/spongepowered/processor/event/ListenerProcessor.java

```diff
@@ -0,0 +1,105 @@
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import javax.tools.Diagnostic;
+
+@SupportedAnnotationTypes(ListenerProcessor.LISTENER_ANNOTATION_CLASS)
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+public class ListenerProcessor extends AbstractProcessor {
+
+    static final String LISTENER_ANNOTATION_CLASS = "org.spongepowered.api.event.Listener";
+    private static final String EVENT_CLASS = Event.class.getName();
+
+    @Override
+    public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        if (ProcessorUtils.contains(annotations, Listener.class)) {
+            for (Element e : roundEnv.getElementsAnnotatedWith(Listener.class)) {
+                if (e.getKind() != ElementKind.METHOD) {
+                    processingEnv.getMessager().printMessage(ERROR, "Invalid element of type " + e.getKind() + " annotated with @Listener", e);
+                    continue;
+                }
+                ExecutableElement method = (ExecutableElement) e;
+
+                Messager msg = this.processingEnv.getMessager();
+                if (method.getModifiers().contains(Modifier.STATIC)) {
+                    msg.printMessage(Diagnostic.Kind.ERROR, "method must not be static", method);
+                }
+                if (!method.getModifiers().contains(Modifier.PUBLIC)) {
+                    msg.printMessage(Diagnostic.Kind.ERROR, "method must be public", method);
+                }
+                if (method.getModifiers().contains(Modifier.ABSTRACT)) {
+                    msg.printMessage(Diagnostic.Kind.ERROR, "method must not be abstract", method);
+                }
+                if (method.getEnclosingElement().getKind().isInterface()) {
+                    msg.printMessage(Diagnostic.Kind.ERROR, "interfaces cannot declare listeners", method);
+                }
+                if (method.getReturnType().getKind() != TypeKind.VOID) {
+                    msg.printMessage(Diagnostic.Kind.ERROR, "method must return void", method);
+                }
+                List<? extends VariableElement> parameters = method.getParameters();
+                if (parameters.isEmpty() || isTypeSubclass(parameters.get(0), EVENT_CLASS)) {
+                    msg.printMessage(Diagnostic.Kind.ERROR, "method must have an Event as its first parameter", method);
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private boolean isTypeSubclass(Element typedElement, String subclass) {
+        Elements elements = processingEnv.getElementUtils();
+        Types types = processingEnv.getTypeUtils();
+
+        TypeMirror event = types.getDeclaredType(elements.getTypeElement(subclass));
+        return types.isAssignable(event, typedElement.asType());
```

Comment:
The order of the arguments for `isAssignable` is wrong here. From the Javadocs:

> returns true if and only if the **first type is assignable to the second**

You want to check if the parameter type extends `Event` which means that you need to check if the parameter is assignable to `Event`. `Event` isn't directly assignable to a subclass of it.